### PR TITLE
Enforce smaller DB transactions in evm transactions querying

### PR DESCRIPTION
**Pull request by LefterisJP**
- Part of https://github.com/rotki/rotki/issues/5222, where we want to make sure DB write transactions do not stay open for long as this creates problems described in the issue
- Remove old forgotten gevent.sleep(0) context switches

### CODE REVIEW

File: rotkehlchen/chain/evm/transactions.py [hunk 1]
```diff
@@ -264,43 +262,44 @@ def _query_and_save_internal_transactions_for_range(
...
+                        write_cursor=write_cursor,
                         transactions=[internal_tx],
                         relevant_address=None,  # no need to re-associate address
                     )
-
-                    if period.range_type == 'timestamps':
-                        assert location_string, 'should always be given for timestamps'
-                        log.debug(f'Internal {self.evm_inquirer.chain_name} transactions for {address} -> update range {period.from_value} - {timestamp}')  # noqa: E501
+                if period.range_type == 'timestamps':
+                    assert location_string, 'should always be given for timestamps'
```
Comment (yabirgb) [inline rotkehlchen/chain/evm/transactions.py:284]: should we keep assets like this in the code?

Comment (LefterisJP) [reply rotkehlchen/chain/evm/transactions.py:284]: You mean asserts? Some times you can do type ignore, some times you can do asserts. The key point is that asserts dissapear in production.

Comment (yabirgb) [reply rotkehlchen/chain/evm/transactions.py:284]: yes asserts. True, I forgot that it gets removed. Thanks

**Comment by LefterisJP** [comment]
All green, will merge

**Pull request merged**